src/scipp/core/shape.py

```diff
@@ -221,3 +221,56 @@ def transpose(x: VariableLike,
     :return: The absolute values of the input.
     """
     return _call_cpp_func(_cpp.transpose, x, dims if dims is not None else [])
+
+
+def squeeze(x: VariableLike,
+            dims: Optional[Union[List[str], Tuple[str, ...]]] = None) -> VariableLike:
+    """Remove dimensions of length 1.
+
+    This is equivalent to indexing the squeezed dimensions with index 0, that is
+    ``squeeze(x, ['x', 'y'])`` is equivalent to ``x['x', 0]['y', 0]``.
+
+    :param x: Object to remove dimensions from.
+    :param dims: If given, the dimensions to squeeze.
+                 If ``None``, all length-1 dimensions are squeezed.
```

Comment:
Numpy also supports and integer instead of a list? Should we support `squeeze(da, dim='x')`?

Reply:
Would be useful.
But I am not sure if we should have 2 mutually exclusive args. We could do it like numpy and use the signular version for both:
```.py
squeeze(x, dim='x')
squeeze(x, dim=['x', 'y'])
```

Reply:
Since likely the most common case (after not giving any dims) is to provide only a single dim I would 👍  this suggestion.